src/core/config.c

```diff
@@ -119,24 +119,46 @@ static char *get_config_path(void) {
 	prefix[0] = config_home;
 	prefix[1] = SYSCONFDIR "/xdg";
 
-	const char *config[2];
-	config[0] = getenv("XDG_CURRENT_DESKTOP");
-	config[1] = "config";
+	char *xdg_current_desktop = getenv("XDG_CURRENT_DESKTOP");
+	int config_num = 1;
+
+	if (xdg_current_desktop && xdg_current_desktop[0] != '\0') {
+		config_num++;
+		for (size_t i = 0; xdg_current_desktop[i] != '\0'; i++) {
+			if (xdg_current_desktop[i] == ':')
+				config_num++;
+		}
+	}
+	xdg_current_desktop = config_num > 1 ? strdup(xdg_current_desktop) : NULL;
+
+	const char *config[config_num];
```

Comment:
I don't think we actually need to build this list? Can't we just call `strtok` unconditionally in the loop that calls `config_path`?

Reply:
we would have to strdup getenv("XDG_CURRENT_DESKTOP") for every prefix in the loop, but that should also work

Reply:
Maybe swap the outer and inner loops to avoid this?

Reply:
then we would alternate between user and system config. I would prefere to first check all possible user configs and then the system ones.

Reply:
I changed it to that. It looks a little bit cleaner. Choose which one you prefer and we can use that version